packages/common/base/src/types/Crossmint.ts

```diff
@@ -0,0 +1,20 @@
+import { validateAPIKey } from "@/apiKey";
+
+export type Crossmint = {
+    apiKey: string;
+    jwt?: string;
+    overrideBaseUrl?: string;
+};
+
+export function createCrossmint(config: Crossmint): Crossmint {
+    const { apiKey, jwt, overrideBaseUrl } = config;
+    const apiKeyValidationResult = validateAPIKey(apiKey);
```

Comment:
I'm curious about the security concerns of validating the API Key client side. I know that we are not exposing any secret, but I wonder if we are making it easier to malicious attackers.

I asked chatGPT, which said:

```
Exposing the validateAPIKey function, which includes decoding and validating API keys, in a package that can be imported by external developers can be considered a bad practice due to several security concerns. Here are the main reasons why:

Security Concerns
Reverse Engineering:
By providing the decoding and validation logic, you make it easier for someone to reverse engineer how your API keys are structured and validated. This can lead to unauthorized access if someone figures out how to generate valid API keys.

Signature Verification:
If someone understands how the signature verification works, they might attempt to create fake API keys that pass the validation. This can compromise the security of your API.

Public Key Exposure:
Exposing the public keys used for signature verification can be a security risk if not managed properly. Even though public keys are meant to be public, knowing the exact keys used for validation can aid in reverse engineering efforts.
```

and the first suggestion was to make it server-side

Reply:
thats just not really true, they would need our private key to make a fake api key

thats like saying are blockchain explorers insecure bc they show the signatures of a transaction...

Reply:
if theres truly a malicious attacker, they would easily be able to figure it out anyways